index.js

```diff
@@ -609,3 +620,7 @@ function shouldForceRelaying (code) {
     (code === 'HOLEPUNCH_DOUBLE_RANDOMIZED_NATS') ||
     (code === 'REMOTE_NOT_HOLEPUNCHABLE')
 }
+
+function shouldBan (err) {
+  return !!err && err.name === 'HypercoreError' && err.code === 'INVALID_OPERATION'
```

Comment:
Aren't we leaking abstractions from hypercore into hyperswarm here? Thinking out loud, but wondering if we can make this a responsibility of the replicate logic (on corestore), that way it's scoped to a part of the stack that knows about hypercore. And we could do something similar for handlers that setup RPC etc.

Reply:
We'd still be leaking abstractions with this approach, just from hyperswarm to corestore in this case, so it's not a good solution.